git-branchless-lib/src/git/status.rs

```diff
@@ -200,7 +200,8 @@ impl TryFrom<&[u8]> for StatusEntry {
                 r#"[\w.]+ "#,                                                // Submodule state.
                 r#"(\d{6} ){2,3}(?P<working_copy_filemode>\d{6}) "#,         // HEAD, Index, and Working Copy file modes;
                                                                              // or stage1, stage2, stage3, and working copy file modes.
-                r#"([\w\d]+ ){2,3}"#,                                        // HEAD and Index object IDs, and optionally the rename/copy score.
+                r#"([[a-f]\d]+ ){2,3}([CR]\d{1,3} )?"#,                      // HEAD and Index object IDs, and optionally the rename/copy score;
+                                                                             // or stage1, stage2, and stage3 object IDs.
```

Comment:
There are numerous ways to update this pattern that could make this less likely to happen, but I don't think we can really eliminate it so long as we use 1 regex for all these porcelain status types. I go into some of them in the PR/commit description.

In particular, the git docs at https://git-scm.com/docs/git-status#_porcelain_format_version_2 state that the `-z` flag (which is used here) causes path names not to be quoted, which ends up leaving the path name a little ambiguous for us.

Reply:
Oops, this is unfortunate. Yeah, refactoring it further sounds non-trivial, so we can hack around it for now.

Reply:
Thanks. NOTE that I updated this line before merge because I had mistakenly doubled the square brackets. This is the change I made just now (`[[a-f]\d]+` → `[a-f\d]+`:
```diff
-                r#"([[a-f]\d]+ ){2,3}([CR]\d{1,3} )?"#,
+                r#"([a-f\d]+ ){2,3}([CR]\d{1,3} )?"#,